tests/PKSim.Tests/Infrastructure/SnapshotSerializerSpecs.cs

```diff
@@ -0,0 +1,106 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using OSPSuite.BDDHelper;
+using OSPSuite.BDDHelper.Extensions;
+using OSPSuite.Utility;
+using PKSim.Core.Snapshots;
+using PKSim.Core.Snapshots.Services;
+using PKSim.Infrastructure.Serialization.Json;
+
+namespace PKSim.Infrastructure
+{
+   public abstract class concern_for_SnapshotSerializer : ContextSpecificationAsync<ISnapshotSerializer>
+   {
+      protected Parameter _parameter;
+      protected string _fileName;
+      protected Parameter _deserialiedParameter;
+
+      public override async Task GlobalContext()
+      {
+         await base.GlobalContext();
+         _fileName = FileHelper.GenerateTemporaryFileName();
+      }
+
+      protected override Task Context()
+      {
+         _parameter = new Parameter();
+         sut = new SnapshotSerializer();
+         return _completed;
+      }
+
+      protected override async Task Because()
+      {
+         await sut.Serialize(_parameter, _fileName);
+         _deserialiedParameter = (await sut.DeserializeAsArray(_fileName, typeof(Parameter))).Cast<Parameter>().First();
+      }
+
+      public override async Task GlobalCleanup()
+      {
+         await base.GlobalCleanup();
+         FileHelper.DeleteFile(_fileName);
+      }
+   }
+
+   public class When_serializing_a_parameter_with_a_value_not_set : concern_for_SnapshotSerializer
+   {
+      protected override async Task Context()
+      {
+         await base.Context();
+         _parameter.Value = null;
+      }
+
+      [Observation]
+      public void should_deserialize_the_parmaeter_with_a_value_also_not_set()
+      {
+         _deserialiedParameter.Value.ShouldBeNull();
+      }
+   }
+
+   public class When_serialziing_a_parameter_with_a_value_set_with_a_mantis_of_length_less_than_the_precision : concern_for_SnapshotSerializer
+   {
+      protected override async Task Context()
+      {
+         await base.Context();
+         _parameter.Value = 1.123456789;
+      }
+
+      [Observation]
+      public void should_not_round_the_number()
+      {
+         _deserialiedParameter.Value.ShouldBeEqualTo(_parameter.Value);
+      }
+   }
+
+   public class When_serializing_a_parameter_whose_value_was_tainted_with_numerical_noise : concern_for_SnapshotSerializer
+   {
+      protected override async Task Context()
+      {
+         await base.Context();
+         _parameter.Value = 449.9999999999991;
```

Comment:
This is the kind of stuff that this PR aims at solving

Reply:
>Can you think of any more tests that I should write?

Maybe smthg in scientific notation? E.g. 199.999999998E-2